pkg/utils/utils.go

```diff
@@ -204,6 +206,15 @@ func HasDpdkDriver(pciAddr string) (bool, error) {
 			return true, nil
 		}
 	}
+
+	/* If there is a vdpa device associated with the device and it is bound to
+	vhost-vdpa driver, it shall be treated as a dpdk device
+	*/
+	if vdpaDev, err := kvdpa.GetVdpaDeviceByPci(pciAddr); err != nil &&
```

Comment:
Is it 1:1 mapping between VF and vDPA dev? I remembered there was discussion about `1 pd => n vdev` where pd is anchor device like mlx5_core.vdpa

Reply:
currently i think its 1:1 (and is statically created for mlx5 vdpa) but:

1. there is work to make it one to many
2. dynamic creation, adding uapi to kernel, vdpa device creation using vdpa tool (iproute2)

Reply:
Right. It is unclear to me how 1:N mapping would be handled by the SR-IOV Device Plugin.
If it does, it could save the detail information (since the PCIAddress is no longer unique) in the device-info-spec file which the CNI could read.
But I guess that's work for the future?

Reply:
yes, this will be for future work.